src/hooks/usePollutedLocationForm.tsx

```diff
@@ -0,0 +1,137 @@
+import { useEffect, useState } from "react";
+import { createPollutedLocation } from "../backEndClient";
+import { ApiRequest } from "../types/backEnd/ApiRequest";
+import {
+  PollutedLocationCreateForm,
+  toPollutedLocationCreateRequest,
+} from "../types/backEnd/PollutedLocationCreateRequest";
+import { mapToPollutedLocation } from "../types/backEnd/PollutedLocationResponse";
+import PollutedLocation, { severityLevels } from "../types/PollutedLocation";
+import { validate } from "../types/Validated";
+import { isInteger, minNumber } from "../utils/validationFunctions";
+
+export interface PollutedLocationFormProps {
+  coordinates: google.maps.LatLngLiteral;
+  setShowCenterMarker: (newValue: boolean) => void;
+  addCreatedPollutedLocation: (newLocation: PollutedLocation) => void;
+}
+
+const usePollutedLocationForm = ({
+  coordinates,
+  setShowCenterMarker,
+  addCreatedPollutedLocation,
+}: PollutedLocationFormProps) => {
+  const [formData, setFormData] = useState<PollutedLocationCreateForm>({
+    radius: {
+      value: 5,
+      errors: [],
+      validationFunctions: [
+        (newValue) => isInteger(newValue),
+        (newValue) => minNumber(newValue, 1),
+      ],
+    },
+    severity: "low",
+    location: {
+      coordinates: {
+        latitude: coordinates.lat,
+        longitude: coordinates.lng,
+      },
+    },
+  });
+
+  const isFormValid = () => {
+    return formData.radius.errors.length === 0;
+  };
+
+  const [request, setRequest] = useState<
+    ApiRequest<PollutedLocation> | undefined
+  >();
+
+  const resetRequest = () => {
+    setRequest(undefined);
+    setShowCenterMarker(true);
+  };
+
+  useEffect(() => {
+    setShowCenterMarker(true);
+
+    return () => {
+      setShowCenterMarker(false);
+    };
+  }, []);
+
+  useEffect(() => {
+    setFormData((previousState) => ({
+      ...previousState,
+      location: {
+        ...previousState.location,
+        coordinates: {
+          latitude: coordinates.lat,
+          longitude: coordinates.lng,
+        },
+      },
+    }));
+  }, [coordinates]);
+
+  const handleSeverityOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newValue = severityLevels.find((l) => l === e.target.value);
+
+    if (!newValue) return;
+
+    setFormData((previousState) => ({
+      ...previousState,
+      severity: newValue,
+    }));
+  };
+
+  const handleRadiusOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData((previousState) => ({
+      ...previousState,
+      radius: validate(formData.radius, +e.target.value),
```

Comment:
would it be difficult to change, so that the new radius value could be blank and not necessarily start with 0? in such case, if we want to for example write 12, we end up having 012, as the 0 is its initial value.

Reply:
It's coming from the fact that in the form state we have a type `number` (it has to have some sort of a number value, 0 is default) and not `number?`. I will check if that's hard to support

Reply:
Changes pushed

![GIF 2022-11-28 19-19-14](https://user-images.githubusercontent.com/29711974/204340855-f50b0030-5fc9-4e06-aaf4-11207957a423.gif)

Reply:
imo that's better. thanks